### PR TITLE
added search method for Song class, implemented pandas csv reader

### PR DESCRIPTION
**Pull request by BorisBabaian**

### CODE REVIEW

File: services/scripts.py [hunk 1]
```diff
@@ -1,43 +1,50 @@
...
+            if request in song.track:
+                _result.append(song)
+        return _result
+    
+        
+            
+            
+            
+            
+
```
Comment (D1ffic00lt) [inline services/scripts.py:47]:
```suggestion
```
just a cosmetic change x2

File: services/scripts.py [hunk 2]
```diff
@@ -1,43 +1,50 @@
 # Music recommendation application
-import csv
-
+import pandas as pd
+from collections import defaultdict
 from cfg import TableIndexes
```
Comment (D1ffic00lt) [inline services/scripts.py:4]:
```suggestion
import pandas as pd

from collections import defaultdict

from cfg import TableIndexes
```
just a cosmetic change

File: services/scripts.py [hunk 3]
```diff
@@ -1,43 +1,50 @@
...
-
-        return Song(row)
+        self._storage = defaultdict(lambda: [])
+        df = pd.read_csv(filename, index_col=0)
+        
+        for _, row in df.iterrows():
+            self.add_song(
+                Song(row)       
+            )
+
```
Comment (D1ffic00lt) [inline services/scripts.py:27]:
```suggestion
```

just a cosmetic change x3

**Review by D1ffic00lt — APPROVED** [review]
LGTM, nice work!

**Review by bani2016up — APPROVED** [review]
Amazing code :)
Very approved! God job!

**Pull request merged**